### PR TITLE
feat: [sc-155806] [api-connect plugin] Add a preset for secure username/password/SSO

### PR DESCRIPTION
**Pull request by alexbourret**
Story details: https://app.shortcut.com/dataiku/story/155806

### CODE REVIEW

File: parameter-sets/secure-basic/parameter-set.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+    "meta" : {
+        "label": "Secure basic",
+        "description": "Username / password to be used on one domain name, https with SSL check ON",
```
Comment (JaneBellaiche) [inline parameter-sets/secure-basic/parameter-set.json:4]: Maybe this message can indicates where to define those credentials ?

**Review by JaneBellaiche — COMMENTED** [review]
Authentication "SSO" with no preset gives:
Capture d’écran 2024-04-17 à 21 45 41
Could we improve the error message ? secure_token is not clear as it does not appear in the UI 

ALso, if the credentials are not set in the profile settings, we get : 
`Test failed: User 'admin' does not have credentials for plugin credential request '"PLUGIN","api-connect","secure-basic","Preset 3","secure_token"' ` which is also not very clear imo, we could indicate how to fix it ?

File: custom-recipes/api-connect/recipe.json [hunk 2]
```diff
@@ -254,6 +281,7 @@
             "name": "ignore_ssl_check",
             "label": "Ignore SSL check",
             "type": "BOOLEAN",
+            "visibilityCondition": "model.auth_type!='secure_oauth' && model.auth_type!='secure_basic'",
```
Comment (JaneBellaiche) [inline custom-recipes/api-connect/recipe.json:284]: Oups, one boolean has been forgotten here,  could you change this one too? 
Capture d’écran 2024-04-18 à 16 49 27

File: python-lib/rest_api_client.py [hunk 3]
```diff
@@ -236,3 +223,15 @@ def enforce_throttling(self):
 
     def get_metadata(self):
         return self.metadata
+
+    def assert_secure_domain(self, url):
+        if not self.secure_domain:
+            return
+        if self.ignore_ssl_check:
+            raise RestAPIClientError("SSL check cannot be desactivated when using this secure preset")
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-lib/rest_api_client.py:225]: 👍 I leave the assertion here but I've added a visibility condition on the check button.

File: custom-recipes/api-connect/recipe.json (hunk 2 above)
Comment (alexbourret) [reply custom-recipes/api-connect/recipe.json:284]: Fixed in [7bc5186](https://github.com/dataiku/dss-plugin-api-connect/pull/40/commits/7bc51861b69b8161c4328d91c04ca4e5dd917a74)

**Review by JaneBellaiche — APPROVED** [review]

**Pull request merged**